tree/tree/src/TTree.cxx

```diff
@@ -6899,7 +6945,7 @@ Long64_t TTree::Merge(TCollection* li, Option_t *options)
 
       Long64_t nentries = tree->GetEntries();
       if (nentries == 0) continue;
-
+      ImportBranches(tree);
```

Comment:
This is a significant new feature and should probably not be enabled by default.  In addition (compare to `RNTuple` which has a similar feature) we do not offer a clear way to select the value used for backfillin. (It is not clear if the current code use a 'default' value, the first value from the new tree or a random value).